include/cutlass/gemm/collective/xe_mma.hpp

```diff
@@ -182,45 +201,44 @@ struct CollectiveMma<MainloopIntelPVC<Stages>, TileShape_, ElementA_, StrideA_,
     (void)residue_mnk;
     (void)thread_idx;
     (void)smem_buf;
-
-    auto tiled_copy_a = make_xe_2d_copy(atom_load_A{}.with(mainloop.mA), Layout<Shape<_1, Int<SubgroupSize>>>{});
-    auto tiled_copy_b = make_xe_2d_copy(atom_load_B{}.with(mainloop.mB), Layout<Shape<_1, Int<SubgroupSize>>>{});
-
-    auto tiled_prefetch_a = XE_Prefetch_A{mainloop.mA};
-    auto tiled_prefetch_b = XE_Prefetch_B{mainloop.mB};
-
-    // Partition the copying of A and B tiles across the threads
-    auto thr_copy_A = tiled_copy_a.get_slice(thread_idx);
-    auto thr_copy_B = tiled_copy_b.get_slice(thread_idx);
+    
+    auto thr_copy_A = mainloop.copy_A.get_slice(thread_idx);
+    auto thr_copy_B = mainloop.copy_B.get_slice(thread_idx);
 
     // Instantiate the MMA object and get thread slice
     TiledMma tiled_mma;
-    auto thr_mma = tiled_mma.get_slice(thread_idx);
-
-    // Partition fragment
-    Tensor fragment_A = thr_mma.partition_fragment_A(gA(_, _, 0));
-    Tensor fragment_B = thr_mma.partition_fragment_B(gB(_, _, 0));
-
-    // Retile for copy
-    Tensor copy_tCrA = thr_copy_A.retile_D(fragment_A);
-    Tensor copy_tCrB = thr_copy_B.retile_D(fragment_B);
-
-    // Retile for cute::gemm
-    Tensor mma_tCrA = thr_copy_A.retile_MMA(thr_mma, fragment_A);
-    Tensor mma_tCrB = thr_copy_B.retile_MMA(thr_mma, fragment_B);
+    // To make all work items in a subgroup have the same global tensors pass in the index of work item 0 in each subgroup
+    auto sg = syclcompat::get_nd_item<1>().get_sub_group();
+    auto first_thread_in_sg_idx = sg.get_group_linear_id() * DispatchPolicy::SubgroupSize;
+    auto thr_mma = tiled_mma.get_slice(first_thread_in_sg_idx);
```

Comment:
Could you put a `TODO(Codeplay)` to deal with this a little more tidily?

Reply:
done